site/en/docs/extensions/mv3/storage-and-cookies/index.md

```diff
@@ -0,0 +1,124 @@
+---
+layout: "layouts/doc-post.njk"
+title: "Storage and Cookies"
+seoTitle: "Chrome Extensions storage and cookies"
+date: 2023-08-01
+description: Overview of how web storage APIs and cookies work in extensions
+---
+
+Extensions can store cookies and access web storage APIs similar to a normal website. However, in
+some cases these behave differently in extensions.
+
+This article does not cover extension specific APIs. For more on these, see
+[`chrome.storage`][chrome-storage-api].
+
+## Storage {: #storage }
+
+It is often desirable to use existing web storage APIs in extensions. This section explores the
+behavior of these APIs in an extension context, which can sometimes differ to how they behave on the
+web.
+
+### Persistence {: #storage-persistence }
+
+Extension storage is not cleared when a user uses the ["Clear Browsing Data"][clear-browsing-data]
+feature. This includes any data stored using web storage APIs (such as
+[Local Storage][local-storage] and [IndexedDB][indexeddb]).
+
+By default, extensions are subject to the normal quota restrictions on storage, which can be checked
+using the `navigator.storage.estimate()` method. Storage can also be evicted under heavy memory
+pressure, although this is rare. To avoid this:
+
+- Request the `unlimitedStorage` permission, which affects both extension and web storage APIs and
+exempts extensions from both quota restrictions and eviction.
+- Use [`navigator.storage.persist()`][storage-persist] for protection against eviction.
+
+Extension storage is shared across the extension's origin, such as the associated service worker,
+any extension pages including popups and the side panel, and offscreen documents. In content
+scripts, calling web storage APIs accesses data from the origin the content script is injected on
+and not the extension.
+
+### Access in service workers {: #storage-in-service-workers }
+
+Some web storage APIs like [IndexedDB][indexeddb] are accessible in service workers. However,
+[Local Storage][local-storage] and [Session Storage][session-storage] are not.
+
+Instead, use an [offscreen document][offscreen]. For example, you could use the following to migrate
+data when updating from Manifest V2 to Manifest V3:
+
+1. In the extension service worker check [`chrome.storage`][chrome-storage-api] for your data.
+1. If your data isn't found, [create][create-offscreen] an offscreen document and call
+[`sendMessage()`][send-message] to start a conversion routine. Read the data from the web storage
+APIs and use message passing to return it to the service worker.
+1. In the service worker, write the data to a new location.
+
+### Partitioning {: #storage-partitioning }
+
+Partitioning is where keys are introduced for stored data to limit where it can be accessed.
+
+To prevent certain types of cross-site tracking, [storage partitioning][storage-partitioning]
+introduces changes to how partitioning keys are defined. In practice, this means that if site A
```

Comment:
```suggestion
specifies how partitioning keys are defined. In practice, this means that if site A
```

Reply:
This text will be on our site for years to come. The language shouldn't sound forever like the feature is new.

Reply:
I understand that general philosophy but I think it might make sense here. Storage partitioning is the name given to the work to partition storage. Once that work is fully rolled out, it will just be how the web works.

Reply:
I've tweaked the wording here to hopefully make it a lot better. Let me know if that sounds good.